mlir/lib/Dialect/MIOpen/Tuning/GridwiseGemmParams.cpp

```diff
@@ -884,48 +434,6 @@ LogicalResult PopulateParamsXDL::getKBlocks(ConvolutionContext &ctx,
                             params.gemmKPack, ctx.num_cu, gemmKBlocks);
 }
 
-LogicalResult PopulateParamsXDL::calculateGemmABlockCopyPerformanceParameters(
-    const InitParamsXDL &param, ConvolutionContext &ctx,
-    DerivedParams &derived) {
-  int64_t blockSize = obtainBlockSize(param, waveSize);
-  return calculateInputDerivedParams(param, blockSize, ctx, true, derived);
-}
-
-LogicalResult PopulateParamsXDL::calculateGemmBBlockCopyPerformanceParameters(
-    const InitParamsXDL &param, ConvolutionContext &ctx,
-    DerivedParams &derived) {
-  int64_t blockSize = obtainBlockSize(param, waveSize);
-  return calculateInputDerivedParams(param, blockSize, ctx, false, derived);
-}
-
-LogicalResult PopulateParamsXDL::calculateLdsNumberOfByte(
-    const InitParamsXDL &param, const ConvolutionContext &ctx,
-    DerivedParams gemmADerived, DerivedParams gemmBDerived, size_t &ldsSize) {
-
-  int64_t threadGemmDataPerRead_GemmM =
-      param.gemmMPerBlock / gemmADerived.clusterLenGemmPos2;
-  int64_t threadGemmDataPerRead_GemmN =
-      param.gemmNPerBlock / gemmBDerived.clusterLenGemmPos2;
-
-  const auto max_lds_align =
-      math_util::lcm(threadGemmDataPerRead_GemmM, threadGemmDataPerRead_GemmN);
-
-  const auto a_block_space =
-      param.gemmKPerBlock *
-      math_util::integer_least_multiple(param.gemmMPerBlock, max_lds_align);
-  const auto b_block_space =
-      param.gemmKPerBlock *
-      math_util::integer_least_multiple(param.gemmNPerBlock, max_lds_align);
-
-  ldsSize = (a_block_space + b_block_space) * sizeof(float);
-
-  if (ldsSize > 64 * 1024) {
-    return failure();
-  }
```

Comment:
Note, previously the assumption was:
 - we assume lds size must be counted as fp32 type
 - The upper bound of the size was statically computed as 64 * 1024

For now I'm bringing the same logic over such that this PR try its best to maintain equivalent coverage. I'd file an additional ticket to document/schedule this such that this is data type and device dependent.